### ISSUE TITLE
Intermittently failing test: BrowserAwesomeBar cancels previous jobs

### ISSUE THREAD

**Issue by grigoryk** [mozilla-mobile/android-components#1300]
```
mozilla.components.browser.awesomebar.BrowserAwesomeBarTest > BrowserAwesomeBar cancels previous jobs if onInputStarted gets called again FAILED
    java.lang.AssertionError at BrowserAwesomeBarTest.kt:170
```

See also #1288.

**Comment by pocmo** [comment]
When re-building the Docker image I hit this error every time.

**Comment by csadilek** [comment]
This should be fixed as well by https://github.com/mozilla-mobile/android-components/commit/4b114bf8beb6d0eecf93abdcf5e9627c19dfa88c

Let's close and reopen in case we're seeing it again.

**Comment by csadilek** [comment]
Seems this one is caused by something else. Reopening...

**Comment by pocmo** [comment]
While debugging this I noticed that the child coroutines do not get cancelled in some situations. Will investigate a bit more.

**Comment by pocmo** [comment]
When running all tests in parallel locally (and my notebook becoming quite slow) I got this intermittent too.

**Comment by dector** [comment]
Can we suggest that this was fixed in #3124?

**Comment by pocmo** [comment]
:+1:
